packages/vxrn/src/exports/prebuild.ts

```diff
@@ -3,60 +3,191 @@ import path from 'node:path'
+          await (async () => {
+            const found = await detectPackageManager()
+            switch (true) {
+              case found.bun:
+                return `bun add -D @react-native-community/template`
+              case found.pnpm:
+                return `pnpm install -D @react-native-community/template`
+              case found.yarn:
+                return `yarn add -D @react-native-community/template`
+              default:
+                return `npm install -D @react-native-community/template`
+            }
+          })()
+        throw new Error(
+          '"@react-native-community/template" is not found in package.json, please install "@react-native-community/template" as dev dependency:\n' +
+            `\`${installCommand}\``
+        )
+      }
     } catch (error) {
-      console.error('Error updating package.json', error)
+      throw new Error('package.json checks are failing:\n' + error)
     }
 
-    // Remove the ios/.xcode.env.local file as it's causing problems `node: No such file or directory` during build
+    // any because we don't have access react-native config
+    let appConfig: any
     try {
-      FSExtra.removeSync(path.join(root, 'ios', '.xcode.env.local'))
-    } catch (e) {
-      // ignore
+      const appConfigPath = path.resolve('app.json')
+      const appConfigContents = await FSExtra.readFile(appConfigPath, 'utf8')
+      appConfig = JSON.parse(appConfigContents)
+    } catch (error) {
+      throw new Error('app.json checks are failing:\n' + error)
     }
-
-    if (!platform || platform === 'ios') {
-      console.info(
-        'Run `open ios/*.xcworkspace` in your terminal to open the prebuilt iOS project, then you can either run it via Xcode or archive it for distribution.'
-      )
+    if (platform == 'ios' || !platform) {
+      if (!doesIOSExist) {
+        await generateForPlatform(root, 'ios', appConfig)
+      } else {
+        console.error('ios/ folder already exists')
+      }
     }
-    if (!platform || platform === 'android') {
-      console.info(
-        '`cd android` then `./gradlew assembleRelease` or `./gradlew assembleDebug` to build the Android project. Afterwards, you can find the built APK at `android/app/build/outputs/apk/release/app-release.apk` or `android/app/build/outputs/apk/debug/app-debug.apk`.'
-      )
+    if (platform == 'android' || !platform) {
+      if (!doesAndroidExist) {
+        await generateForPlatform(root, 'android', appConfig)
+      } else {
+        console.error('android/ folder already exists')
+      }
     }
-  } catch (e) {
-    console.error(`Failed to prebuild native project: ${e}\nIs "expo" listed in your dependencies?`)
+  }
+  if (!platform || platform === 'ios') {
+    console.info(
+      'Run `open ios/*.xcworkspace` in your terminal to open the prebuilt iOS project, then you can either run it via Xcode or archive it for distribution.'
```

Comment:
This instruction here may be out of place since prebuild without expo won't run `pod install` automatically (while expo will), and there will be no `ios/*.xcworkspace` at this time.